SimplifiedTests/NYPLAxisTests/NYPLAxisDownloadProgressTests.swift

```diff
@@ -0,0 +1,312 @@
+    return axisDownloadProcess
+  }()
+
+  override func tearDown() {
+    super.tearDown()
+    sut.test_reset()
+  }
+
+  func testDownloadingFixedWeightItemsShouldUpdateDownloadProgress() {
+    sut.addFixedWeightTask(with: licenseURL, weight: 0.05)
+    sut.addFixedWeightTask(with: containerURL, weight: 0.05)
+
+    XCTAssertEqual(progressListener.currentProgress, 0.0)
+
+    sut.didFinishTask(with: licenseURL)
+    // 5 % downloaded
+    XCTAssertEqual(progressListener.currentProgress, 0.05)
+
+    sut.addFixedWeightTask(with: encryptionURL, weight: 0.4)
+    // 5 % downloaded
+    XCTAssertEqual(progressListener.currentProgress, 0.05)
+
+    sut.didFinishTask(with: encryptionURL)
+    // 45 % downloaded
+    XCTAssertEqual(progressListener.currentProgress, 0.45)
+  }
+
+  func testDownloadingFlexibleWeightItemShouldUpdateDownloadProgress() {
+    sut.addFixedWeightTask(with: licenseURL, weight: 0.05)
+    sut.addFixedWeightTask(with: containerURL, weight: 0.05)
+    sut.addFlexibleWeightTask(with: chapter1URL)
+    sut.addFlexibleWeightTask(with: chapter2URL)
+
+    XCTAssertEqual(progressListener.currentProgress, 0.0)
+
+    sut.didFinishTask(with: chapter1URL)
+    // 45 % downloaded
+    XCTAssertEqual(progressListener.currentProgress, 0.45)
+
+    sut.didFinishTask(with: chapter2URL)
+    // 90% downloaded
+    XCTAssertEqual(progressListener.currentProgress, 0.9)
+  }
+
+  func testAddingMoreItemsShouldNotChangeCurrentDownloadProgress() {
+    sut.addFixedWeightTask(with: licenseURL, weight: 0.05)
+    sut.addFixedWeightTask(with: containerURL, weight: 0.05)
+    sut.addFlexibleWeightTask(with: chapter1URL)
+    sut.addFlexibleWeightTask(with: chapter2URL)
+
+    XCTAssertEqual(progressListener.currentProgress, 0.0)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.45)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.45)
+
+    sut.didFinishTask(with: chapter1URL)
+    // 45 % downloaded
+    XCTAssertEqual(progressListener.currentProgress, 0.45)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.45)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.45)
+
+    sut.addFlexibleWeightTask(with: encryptionURL)
+    // 45 % downloaded
+    XCTAssertEqual(progressListener.currentProgress, 0.45)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.45)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.45)
```

Comment:
Correct me if I am wrong, we now have two unfinished flexible weight tasks, should the weight of `chapter2URL` be 0.225 ((1 - 0.05 - 0.05 - 0.45) / 2)?

Reply:
You sir are absolutely right. Not only was there a problem with my math, but there was also a bug 🤮  in my logic. Thank you for pointing it out. Fixed.